packages/office-ui-fabric-react/src/utilities/decorators/withViewport.tsx

```diff
@@ -84,35 +84,36 @@ export function withViewport<TProps extends { viewport?: IViewport }, TState>(
       // particularly bad when skipViewportMeasures is set when optimizing fixed layout lists.
       // It will measure and update and re-render the entire list after list is fully rendered.
       // So fallback to listen to resize event when skipViewportMeasures is set.
-      if (!skipViewportMeasures && this._isResizeObserverAvailable()) {
-        this._registerResizeObserver();
-      } else {
-        this._events.on(win, 'resize', this._onAsyncResize);
-      }
-
       if (!skipViewportMeasures) {
+        if (this._isResizeObserverAvailable()) {
+          this._registerResizeObserver();
+        } else {
+          this._events.on(win, 'resize', this._onAsyncResize);
+        }
+
         this._updateViewport();
       }
     }
 
-    public componentDidUpdate(newProps: TProps) {
```

Comment:
I have no idea why people didn't notice that the contract was wrong here for so long.